src/test/regress/expected/with_clause_optimizer.out

```diff
@@ -2179,3 +2179,174 @@ WITH RECURSIVE r1 AS (
 )
 SELECT * FROM r1 LIMIT 1;
 ERROR:  joining nested RECURSIVE clauses is not supported
+--
+-- Test various SELECT statements over DML operations
+--
+create table with_dml(i integer, j integer) distributed by(i);
+create table with_dml_dr(i integer, j integer) distributed replicated;
+-- Test select over insert queries. Gather motion should be there, no matter
+-- what type of distribution is used. Before fix, motion was missed for
+-- distributed replicated table, which caused count(*) to return zero rows.
+explain (costs off)
+with cte as (
+    insert into with_dml select i, i * 100 from generate_series(1,5) i
+    returning i
+) select count(*) from cte;
+                                  QUERY PLAN                                  
+------------------------------------------------------------------------------
+ Aggregate
+   ->  Gather Motion 3:1  (slice2; segments: 3)
+         ->  Aggregate
+               ->  Subquery Scan on cte
+                     ->  Insert on with_dml
+                           ->  Redistribute Motion 1:3  (slice1; segments: 1)
+                                 Hash Key: i.i
+                                 ->  Function Scan on generate_series i
+ Optimizer: Postgres query optimizer
+(9 rows)
+
+with cte as (
+    insert into with_dml select i, i * 100 from generate_series(1,5) i
+    returning i
+) select count(*) from cte;
+ count 
+-------
+     5
+(1 row)
+
+explain (costs off)
+with cte as (
+    insert into with_dml_dr select i, i * 100 from generate_series(1,5) i
+    returning i
+) select count(*) from cte;
+                          QUERY PLAN                           
+---------------------------------------------------------------
+ Aggregate
+   ->  Subquery Scan on cte
+         ->  Explicit Gather Motion 3:1  (slice1; segments: 3)
+               ->  Insert on with_dml_dr
+                     ->  Function Scan on generate_series i
+ Optimizer: Postgres query optimizer
+(6 rows)
+
+with cte as (
+    insert into with_dml_dr select i, i * 100 from generate_series(1,5) i
+    returning i
+) select count(*) from cte;
+ count 
+-------
+     5
+(1 row)
+
+-- Test select over update queries. Check no qual pushed down to update from
+-- select, because it's incorrect to push it down in such way. Queries should
+-- be correctly executed without any rewriting error, e.g. rewriteManip.c:1409.
```

Comment:
Probably it is better to specify log message as code at [rewriteManip.c:1409](https://github.com/arenadata/gpdb/blob/2aea19802afbac1d0fa2f4b7c7d7eec5de9e3eaa/src/backend/rewrite/rewriteManip.c#L1409) can change its number later.

> "could not find replacement targetlist entry for attno %d"